src/views/DeploymentsProvider.ts

```diff
@@ -0,0 +1,411 @@
+            const localhostData = await this.fetchLocalhostRoutes(this.localhostPorts);
+            const kubernetesData = await this.fetchKubernetesData();
+
+            for (const [key, value] of localhostData.entries()) {
+                this.localhostData.set(key, value);
+            }
+
+            for (const [key, value] of kubernetesData.entries()) {
+                this.kubernetesData.set(key, value);
+            }
+
+            console.log('Data initialized:', {
+                localhost: this.localhostData,
+                kubernetes: this.kubernetesData
+            });
+
+            this.loading = false; // Mark as no longer loading
+            this.refresh(); // Refresh the UI with loaded data
+        } catch (error) {
+            console.error('Error initializing data:', error);
+            this.loading = false; // Ensure loading is marked as false even on error
+            this.refresh(); // Refresh UI to show the state
+        }
+    }
+
+    // fetch data from Camel Run Console running at defined port
+    private async fetchLocalhostRoutes(ports: number[]): Promise<Map<string, Route[]>> {
+        const deployments: Map<string, Route[]> = new Map();
+        const skippedPorts: number[] = [];
+
+        // there can be multiple running integrations --> fetching all active (existing) ports
+        const fetchPromises = ports.map(async (port) => {
+            try {
+                const response = await fetch(`http://localhost:${port}/q/dev/route`, {
+                    headers: { 'Accept': 'application/json' }
+                });
+
+                if (!response.ok) {
+                    console.warn(`Port ${port} not available: ${response.statusText}`);
+                    skippedPorts.push(port);
+                    return; // Skip this port
+                }
+
+                const parsedFile = await response.json();
+                console.log(`Data from port ${port}:`, parsedFile);
+
+                if (parsedFile) {
+                    for (const jsonRoute of parsedFile.route.routes) {
+                        const route = new Route(
+                            jsonRoute.routeId,
+                            jsonRoute.source,
+                            jsonRoute.from,
+                            jsonRoute.remote,
+                            jsonRoute.state,
+                            jsonRoute.uptime,
+                            jsonRoute.statistics
+                        );
+
+                        const fileNameKey = getBasenameIfAbsolute(route.associatedFile)
+                        deployments.has(fileNameKey)
+                            ? deployments.get(fileNameKey)?.push(route)
+                            : deployments.set(fileNameKey, [route]);
+                    }
+                }
+            } catch (error: any) {
+                if (error.message.includes('fetch failed')) {
+                    // Skip failed ports which are not used at all
```

Comment:
we should not swallow exceptions.
(it is passing here for me when the route is not displayed in the deployments)